### PR TITLE
Create Error Handling 'review' lesson plan

### PR DESCRIPTION
**Pull request by jamisonordway**
# Backend Curriculum PR Template

### Description

- This change adds a new "lesson plan" called Error Handling Review which is meant to supplement students' learning in terms of error handling and Exceptions in Rails.
- Based on an older version of the main Error Handling lesson, which references [House Salad](https://github.com/turingschool-examples/house-salad-7).

### Why are we making this update?

The existing version of the Error Handling lesson had a few problems:
- House Salad repo was referenced in code examples, but never linked
- House Salad examples were out of date compared to the work students have done on this repo recently

The plan is to update these examples to reflect recent versions of House Salad, with another PR to update the main lesson plan with a different exercise/demo.
  
### Type of update

- [ ] Minor update/fix -- no review requested
- [x] Moderate update -- review from Mod Team requested
- [ ] Major update -- review from Backend Team requested

### How will we measure the success of this change? 

Students will get more value out of (and leave positive feedback about) future review sessions for this topic.

### What questions do you have/what do you want feedback on? (optional)

- Memoization-- should we remove this recommendation?
- Serializers
- Amount of logic handled in `SearchFacade`
- any other ideas and suggestions welcome!

### CODE REVIEW

File: module3/lessons/error_handling_review.md [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+  @@service_result ||= CongressService.senate_members
+end
+
+def all_senate_members
+  senate_service[:results][0][:members]
+end
+```
+Isolation helps on a few fronts. It adheres more closely to SRP, and we can properly test this method. We can also take advantage of memoization to ensure our we only ever make one external API call per request.
+
+Because we are memomizing within a class method, we need a class variable to hold the result of our service call. An instance variable or local variable will not memoize properly.
```
Comment (cjsim89) [inline module3/lessons/error_handling_review.md]: Typo `memomizing` => `memoizing`

Comment (cjsim89) [reply module3/lessons/error_handling_review.md]: I feel like memoization could be a whole other topic honestly, and while I like that we introduce it, we're also introducing the concept of a class variable which we haven't done before. Still thinking about this, but I would almost want to link to Rails/Ruby docs with these topics, or make another sub-lesson that explains them deeper. 🤔

**Review by cjsim89 — COMMENTED** [review]
Overall looks good, I like that it focuses on error handling only, and relies on students already knowing what the topic is/why we do it. Plus, it doesn't feature any of the other stuff in the old version of the lesson, a lot cleaner.

File: module3/lessons/error_handling_review.md (hunk 1 above)
Comment (cjsim89) [reply module3/lessons/error_handling_review.md]: Do we also have a completed version of this review on a branch in house-salad-7? If so, can we link it at the bottom under the Checks for Understanding?

Comment (jamisonordway) [reply module3/lessons/error_handling_review.md]: 1) I agree-- after working through this review I don't think these examples of memoization and class variables are valuable, anyway. Especially the memoization in `members_by_last_name`, which can actually cause a result to be returned regardless of whether the user's query matches any data. I'll take out the references to class vars and memoization
2) Yes, there's a completed version on `error_handling_complete`, I'll be sure to add that at the end of the lesson.

**Review by cjsim89 — APPROVED** [review]
👍

**Pull request merged**